### ISSUE TITLE
geary optimization

### ISSUE THREAD

**Issue by ljwolf** [pysal/esda#44]
we should be able to exploit `numpy.subtract.outer` and elementwise matrix multiplication to do the Geary statistic in a much faster fashion than our current list comprehension. Something like:

```python
numerator = (numpy.subtract.outer(y,y)**2).multiply(w.sparse).sum()
```

**Comment by ljwolf** [comment]
or, even better, use `sklearn.metrics.pairwise` and we might be able to get something like to @lanselin's multivariate geary with minimal change (though still might need to fix the permutation inference stuff?)

**Comment by sjsrey** [comment]
> we should be able to exploit `numpy.subtract.outer` and elementwise matrix multiplication to do the Geary statistic in a much faster fashion than our current list comprehension. Something like:
> 
> ```python
> numerator = (numpy.subtract.outer(y,y)**2).multiply(w.sparse).sum()
> ```

the outer will generate a dense nxn matrix though?

**Comment by jGaboardi** [comment]
Closed by #114.
